store/tikv/client.go

```diff
@@ -224,36 +234,58 @@ func NewTestRPCClient() Client {
 	return newRPCClient(config.Security{})
 }
 
-func (c *rpcClient) getConnArray(addr string) (*connArray, error) {
+func (c *rpcClient) getConnArray(addr string, ctype connType) (*connArray, error) {
 	c.RLock()
 	if c.isClosed {
 		c.RUnlock()
 		return nil, errors.Errorf("rpcClient is closed")
 	}
-	array, ok := c.conns[addr]
+	var array *connArray
+	var ok bool
+	switch ctype {
+	case Read:
+		array, ok = c.readConns[addr]
+	case Write:
+		array, ok = c.writeConns[addr]
+	}
 	c.RUnlock()
 	if !ok {
 		var err error
-		array, err = c.createConnArray(addr)
+		array, err = c.createConnArray(addr, ctype)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return array, nil
 }
 
-func (c *rpcClient) createConnArray(addr string) (*connArray, error) {
+func (c *rpcClient) createConnArray(addr string, ctype connType) (*connArray, error) {
 	c.Lock()
 	defer c.Unlock()
-	array, ok := c.conns[addr]
+	var array *connArray
+	var ok bool
+	switch ctype {
+	case Read:
+		array, ok = c.readConns[addr]
+	case Write:
+		array, ok = c.writeConns[addr]
+	}
 	if !ok {
 		var err error
 		connCount := config.GetGlobalConfig().TiKVClient.GrpcConnectionCount
+		if ctype == Read && connCount > 2 {
+			connCount /= 2
```

Comment:
Maybe assign less conns to read? As the results in #12569 shows 3 conn is best for write, 2 conn is best for read.

Reply:
I will make read and write connection count configurable after we have got a proper default value.

Reply:
If `connCount == 3` , `/=2` gets `1`, which is not optimal.

Reply:
I have make point get connection count configurable, and will give a default value after i have finished performance test.